syntax = "proto3";
package Protocol;


message Battery
{
    int32 BatteryPK = 1;
    int32 BAT_BPU_NEGATIVE_CONTACTOR = 2;
    int32 BAT_BPU_POSITIVE_CONTACTOR = 3;
    int32 BAT_BPU_PRE_CONTACTOR = 4;
    int32 BAT_PROTECTION_A = 5;
    int32 BAT_PROTECTION_B = 6;
    string System_time = 7;
}

message Battery_Pack
{
    int32 Battery_Pack = 1;
    int32 BAT_PACK_CURRENT = 2;
    int32 BAT_PACK_SOC = 3;
    int32 BAT_PACK_VOLTAGE = 4;
    int32 BAT_PACK_MAX_TEMP = 5;
    int32 BAT_PACK_MAX_TEMP_SBMS_INDEX = 6;
    int32 BAT_PACK_MIN_TEMP = 7;
    int32 BAT_PACK_MIN_TEMP_SBMS_INDEX = 8;
    int32 BAT_PACK_MAX_CELL_VOL = 9;
    int32 BAT_PACK_MAX_CELL_VOL_SBMS_INDEX = 10;
    int32 BAT_PACK_MIN_CELL_VOL = 11;
    int32 BAT_PACK_MIN_CELL_VOL_SBMS_INDEX = 12;
    string System_time = 13;
}

message BAT_MODULE_0
{
    int32 BatteryMd_0 = 1;
    int32 BAT_MODULE_CELL0_VOL = 2;
    int32 BAT_MODULE_CELL1_VOL = 3;
    int32 BAT_MODULE_CELL2_VOL = 4;
    int32 BAT_MODULE_CELL3_VOL = 5;
    int32 BAT_MODULE_CELL4_VOL = 6;
    int32 BAT_MODULE_CELL5_VOL = 7;
    int32 BAT_MODULE_CELL6_VOL = 8;
    int32 BAT_MODULE_CELL7_VOL = 9;
    int32 BAT_MODULE_CELL8_VOL = 10;
    int32 BAT_MODULE_CELL9_VOL = 11;
    int32 BAT_MODULE_CELL10_VOL = 12;
    int32 BAT_MODULE_CELL11_VOL = 13;
    string System_time = 14;
}

message BAT_MODULE_1
{
    int32 BatteryMd_1 = 1;
    int32 BAT_MODULE1_CELL0_VOL = 2;
    int32 BAT_MODULE1_CELL1_VOL = 3;
    int32 BAT_MODULE1_CELL2_VOL = 4;
    int32 BAT_MODULE1_CELL3_VOL = 5;
    int32 BAT_MODULE1_CELL4_VOL = 6;
    int32 BAT_MODULE1_CELL5_VOL = 7;
    int32 BAT_MODULE1_CELL6_VOL = 8;
    int32 BAT_MODULE1_CELL7_VOL = 9;
    int32 BAT_MODULE1_CELL8_VOL = 10;
    int32 BAT_MODULE1_CELL9_VOL = 11;
    int32 BAT_MODULE1_CELL10_VOL = 12;
    int32 BAT_MODULE1_CELL11_VOL = 13;
    string System_time = 14;
}

message BAT_MODULE_2
{
    int32 BatteryMd_2 = 1;
    int32 BAT_MODULE2_CELL0_VOL = 2;
    int32 BAT_MODULE2_CELL1_VOL = 3;
    int32 BAT_MODULE2_CELL2_VOL = 4;
    int32 BAT_MODULE2_CELL3_VOL = 5;
    int32 BAT_MODULE2_CELL4_VOL = 6;
    int32 BAT_MODULE2_CELL5_VOL = 7;
    int32 BAT_MODULE2_CELL6_VOL = 8;
    int32 BAT_MODULE2_CELL7_VOL = 9;
    int32 BAT_MODULE2_CELL8_VOL = 10;
    int32 BAT_MODULE2_CELL9_VOL = 11;
    int32 BAT_MODULE2_CELL10_VOL = 12;
    int32 BAT_MODULE2_CELL11_VOL = 13;
    string System_time = 14;
}

message BAT_MODULE_3
{
    int32 BatteryMd_3 = 1;
    int32 BAT_MODULE3_CELL0_VOL = 2;
    int32 BAT_MODULE3_CELL1_VOL = 3;
    int32 BAT_MODULE3_CELL2_VOL = 4;
    int32 BAT_MODULE3_CELL8_VOL = 5;
    int32 BAT_MODULE3_CELL10_VOL = 6;
    int32 BAT_MODULE3_CELL11_VOL = 7;
}

message Environment
{
    int32 Env = 1;
    int32 Wind_speed = 2;
    int32 Wind_direction = 3;
    string System_time = 4;
}

message AIS
{
    int32 AIS_PK = 1;
    string latitude = 2;
    string longitude = 3;
    string System_time = 4;
}

message System_Time
{
    int32 Times = 1;
    string System_time = 2;
}

message MOTOR
{
    int32 MOTOR_PK = 1;
    int32 MT_RPM = 2;
    int32 MT_TORQUE = 3;
    int32 MT_TEMP = 4;
    string System_time = 5;
}

message INVERTER
{
    int32 INVERTER_PK = 1;
    int32 INV_PHASE_A_CURRENT = 2;
    int32 INV_PHASE_B_CURRENT = 3;
    int32 INV_PHASE_C_CURRENT = 4;
    int32 INV_POST_FAULT = 5;
    int32 INV_GATE_DRIVER_BOARD_TEMP = 6;
    int32 INV_MODULE_A_TEMP = 7;
    int32 INV_RUN_FAULT = 8;
    int32 INV_MODULE_B_TEMP = 9;
    int32 INV_MODULE_C_TEMP = 10;
    int32 INV_POWER = 11;
    int32 INV_OUTPUT_VOLTAGE = 12;
    string System_time = 13;
}
